micro-rdk/src/common/conn/server.rs

```diff
@@ -276,8 +304,9 @@ pub struct ViamServer<C, T, CC, D, L> {
     exec: Executor,
     app_connector: C,
     app_config: AppClientConfig,
-    app_client: Option<AppClient>,
+    app_client: Rc<futures_util::lock::Mutex<Option<AppClient>>>,
```

Comment:
The key change: access to the AppClient is now Rc'd and async Mutex'd. Many async tasks can hold a handle to it, each can await having private access to it, and they all hold a consistent view of whether the AppClient managed by the main `serve` loop is currently populated or not.

Reply:
Good point about rw. There isn't an async rwlock in `futures_utils` but I could try https://docs.rs/async-rwlock/latest/async_rwlock/ if we are OK with picking up that dependency and it works out. I'll do that now and see how it goes.

Reply:
Seems like the newer version is https://docs.rs/async-lock/3.3.0/async_lock/